### PR TITLE
fetchAssoc() shouldn't return bool anymore

### PR DESCRIPTION
**Pull request by Faneraiy14**
As you pointed out, since PdoDb uses PDO::ERRMODE_EXCEPTION, a failed query just throws instead of giving back false, so fetchAssoc() can never actually return bool - that part of the type was never really true.

What I did:
- PdoDbResult's constructor now only takes a PDOStatement (no more bool), and fetchAssoc() on both the interface and the class is typed array|null.
- Removed 3 `=== false` checks that PHPStan flagged as dead once the type was fixed (PasswordHashStatsCommand, RefreshTokenRepository, UserSettingsRepository) - they were guarding against something that could never happen.
- One test was mocking fetchAssoc() to return false to test a "query failed" case. That case doesn't exist anymore, so I removed it - the real failure case (an exception getting thrown) is already covered by another test right next to it.
- I went through every other place that calls fetchAssoc() too. A couple of spots (CreditSystem, HistoryRepository) grab a column off the result without checking for null first, but they always check rowCount() right before, so null can't actually happen there. Didn't touch those since nothing's actually broken, just wanted to mention I checked.

phpstan/phpcs/phpunit all green.

### CODE REVIEW

File: src/Db/DbResultInterface.php [hunk 1]
```diff
@@ -5,7 +5,7 @@
 interface DbResultInterface
 {
     /**
-     * @return array|bool|null
+     * @return array|null
      */
     public function fetchAssoc();
```
Comment (sveneld) [inline src/Db/DbResultInterface.php]: now we can add types not in phpdoc but as declaration
public function fetchAssoc(): ?array;

and it is possible to do for all methods here

**Review by sveneld — CHANGES_REQUESTED** [review]

File: src/Db/DbResultInterface.php (hunk 1 above)
Comment (Faneraiy14) [reply src/Db/DbResultInterface.php]: done, switched all three methods to real return types instead of just phpdoc

**Review by sveneld — CHANGES_REQUESTED** [review]
Thanks for picking this up — the direction is right and most of it is clean (honest constructor type, single `null` sentinel, dead `=== false` guards removed, and the `PdoDbTest` proving a failed query throws). One blocker before merge though.

## `fetchAssoc(): ?array` + the retained `rowCount()` gate is a regression

`fetchAssoc()` now declares `: ?array`, but it still gates on `rowCount()` and returns `$this->result->fetch(...)` directly — and `PDOStatement::fetch()` returns `array|false`. On MariaDB `rowCount()` does **not** decrease as you consume the cursor, so in a `while ($row = $result->fetchAssoc())` loop the *terminating* call has `rowCount() > 0` but `fetch()` returns `false` → the method returns `false` against a `?array` return type → **TypeError**.

Empirically reproduced against the real MariaDB (two-row result set, same shape as `NoteRepository::findActiveNotes()`):

```
rowCount() = 2
iter 1 => n=1
iter 2 => n=2
THREW after 2 iterations: TypeError: BikeShare\Db\PdoDbResult::fetchAssoc(): Return value must be of type ?array, false returned
```

The suite stays green only because the one loop of this shape in the repo, `NoteRepository::findActiveNotes()`, currently has no callers and no test — so this is a latent break of the `while (fetchAssoc())` contract, not a live production crash today. Still worth fixing here since it's cheap and the whole point of this change is to make the contract honest.

**Fix** — drop the `rowCount()` gate and normalise `fetch()`'s `false` to `null` (this is exactly what spec 0022 §5 prescribed). Verified it makes the repro loop terminate cleanly:

```php
public function fetchAssoc(): ?array
{
    $row = $this->result->fetch(PDO::FETCH_ASSOC);

    return $row === false ? null : $row;
}
```

While you're there, please add a unit test for the multi-row case (fetch until the set is exhausted, assert the terminating call returns `null`) so this can't regress silently.

## Minor: the unguarded column reads

You correctly flagged that `CreditSystem:161` and `HistoryRepository:229/251/306` read a column straight off `fetchAssoc()` without a null check. That's only safe *because* the `rowCount()` gate lives inside `fetchAssoc()` today; once the gate moves out (fix above), they rely on "this query always returns exactly one row". That's true for those aggregates, but please make the invariant explicit with a short comment (or a local guard) at each site, per spec 0022 §5 — so the next reader doesn't have to re-derive it.

**Review by sveneld — APPROVED** [review]
Verified on a fresh checkout: the rowCount()-gate regression is fixed exactly per spec 0022 §5 (fetch()-first, false→null), the empirical while-loop repro now ends cleanly on real MariaDB, invariant comments added at the four unguarded call sites, and phpstan + phpcs + 483 tests are green. Thanks for the thorough follow-up.
